apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -52,26 +53,80 @@ class AnimationPickerBody extends React.Component {
     spriteLab: PropTypes.bool
   };
 
-  state = {
-    searchQuery: '',
-    categoryQuery: '',
-    currentPage: 0
+  constructor(props) {
+    super(props);
+
+    const initialState = {
+      searchQuery: '',
+      categoryQuery: '',
+      currentPage: 0,
+      libraryManifest: props.spriteLab ? spriteCostumeLibrary : animationLibrary
+    };
+    const results = this.searchAssets(initialState.currentPage, initialState);
```

Comment:
This looks like it should be calling `searchAssets` directly rather than `this.searchAssets`

Reply:
why is that?

Reply:
My thought: this.searchAssets is using `this.state`, but `this.state` hasn't been initialized yet. In theory, we'd never hit the branch where `this.searchAssets` uses `this.state`, but it seems fragile to future changes. 

Additionally, there's a general _feel_ that's making me uncomfortable (maybe code smell?), but I'm not sure if I can put my finger on why. I think this might be it: the constructor is where we would have bound `this.searchAssets` prior to arrow functions. I'm not clear on the order of operations here since we're binding it with the arrow function later, but it's possible that `this` when used within `this.searchAssets` is not what we're expecting. Using a non-static member function in a constructor _feels_ like an odd pattern.

Reply:
i think that makes sense. `this.searchAssets` will always be bound, so that isn't a concern, but agreed about `this.state` not being initialized yet making this a little confusing